app/controllers/products_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ProductsController < ApplicationController
+  def edit
+    @product = Product.find(params[:id])
+  end
+
+  def update
+    @product = Product.find(params[:id])
+    @product.update(product_update_params[:product])
```

Comment:
Is there a reason that in this method you use update, but in markets controller you use the bang❗️ version (update!)?